lib/services.credentials.js

```diff
@@ -58,29 +83,42 @@ bedrock.events.on('bedrock-express.configure.routes', function(app) {
           query['credential.sysState'] = req.query.filter;
         }
         if(req.query.id) {
-          // FIXME: use _getCredentialRecord()
-          store.get(
-            req.user.identity, req.query.id,
-            function(err, credential, meta) {
-              if(err) {
-                return callback(err);
-              }
-              _credentialToFormat(credential, meta, req.query.format, callback);
-            });
-        } else {
-          // FIXME: use _getCredentialRecord() or similar
-          store.getAll(
-            req.user.identity, query, {credential: true, meta: true},
-            function(err, records) {
-              if(err) {
-                return callback(err);
+          return async.waterfall([
+            function(callback) {
+              store.get(req.user.identity, req.query.id, callback);
+            },
+            function(credential, meta, callback) {
+              if(!_checkCredentialParties(req.user.identity, credential)) {
```

Comment:
This isn't quite the right way to do this check. We should be checking using the permission API.